pkg/data/implementations/gcp_remote_url.go

```diff
@@ -0,0 +1,189 @@
+package implementations
+
+import (
+	"context"
+	"time"
+
+	gax "github.com/googleapis/gax-go/v2"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/admin"
+	"golang.org/x/oauth2"
+
+	credentials "cloud.google.com/go/iam/credentials/apiv1"
+	gcs "cloud.google.com/go/storage"
+	"github.com/lyft/flyteadmin/pkg/data/interfaces"
+	"github.com/lyft/flyteadmin/pkg/errors"
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/lyft/flytestdlib/storage"
+	"google.golang.org/api/option"
+	credentialspb "google.golang.org/genproto/googleapis/iam/credentials/v1"
+	"google.golang.org/grpc/codes"
+)
+
+const gcsScheme = "gs"
+
+type iamCredentialsInterface interface {
+	SignBlob(ctx context.Context, req *credentialspb.SignBlobRequest, opts ...gax.CallOption) (*credentialspb.SignBlobResponse, error)
+	GenerateAccessToken(ctx context.Context, req *credentialspb.GenerateAccessTokenRequest, opts ...gax.CallOption) (*credentialspb.GenerateAccessTokenResponse, error)
+}
+
+type gcsClientWrapper struct {
+	delegate *gcs.Client
```

Comment:
hey @honnix I'm not super familiar with the delegate pattern, what's the reason for using it here?

Reply:
is this so you can mock out the bucket like you commented below?

Reply:
Yes. That is the only way I could figure out how to test this fluid API.